dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/test-applications/thirdpartylib/src/com/ibm/ws/jaxrs2x/fat/thirdparty/multipart/UploadServiceImpl.java

```diff
@@ -70,6 +73,37 @@ public Response uploadFile(List<IAttachment> attachments, @Context HttpServletRe
         return Response.ok("file uploaded").build();
     }
 
+    @POST
+    @Path("/uploadFile2")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response uploadFile2(List<IAttachment> attachments, @Context HttpServletRequest request) {
+        for (IAttachment attachment : attachments) {
+            DataHandler handler = attachment.getDataHandler();
+            try {
+                InputStream stream = handler.getInputStream();
+                MultivaluedMap<String, String> map = attachment.getHeaders();
+                System.out.println("fileName2 Here " + getFileName(map));
+                OutputStream out = new FileOutputStream(new File("./" + getFileName(map)));
+                StringBuilder stringBuilder = new StringBuilder();
+                Reader in = new InputStreamReader(stream, StandardCharsets.UTF_8);
+                char[] chars = new char[1024];
+                int charsRead;
+                while((charsRead = in.read(chars, 0, chars.length)) > 0) {
+                    stringBuilder.append(chars, 0, charsRead);
+                    out.write(new String(chars).getBytes("UTF-8"), 0, charsRead);
+                }
+                System.out.println("uploadFile2 stringBuilder.toString(): " + stringBuilder.toString());
+                stream.close();
```

Comment:
Probably you should move the stream.close(), out.flush() and out.close() into a finally{}

Reply:
+1 or use try-with-resources to auto close